src/components/application_manager/rpc_plugins/sdl_rpc_plugin/test/commands/mobile/close_application_request_test.cc

```diff
@@ -0,0 +1,309 @@
+ * disclaimer in the documentation and/or other materials provided with the
+ * distribution.
+ *
+ * Neither the name of the Ford Motor Company nor the names of its contributors
+ * may be used to endorse or promote products derived from this software
+ * without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include <memory>
+#include <string>
+#include <cstdint>
+
+#include "mobile/close_application_request.h"
+
+#include "gtest/gtest.h"
+
+#include "smart_objects/smart_object.h"
+#include "application_manager/smart_object_keys.h"
+#include "application_manager/commands/command_request_test.h"
+#include "application_manager/mock_application_manager.h"
+#include "application_manager/mock_application.h"
+#include "application_manager/mock_rpc_service.h"
+#include "connection_handler/mock_connection_handler.h"
+#include "protocol_handler/mock_session_observer.h"
+#include "utils/macro.h"
+
+namespace test {
+namespace components {
+namespace commands_test {
+namespace mobile_commands_test {
+namespace close_application_request {
+
+using ::testing::_;
+using ::testing::Return;
+using ::testing::ReturnRef;
+using am::commands::MessageSharedPtr;
+namespace am = ::application_manager;
+
+typedef std::shared_ptr<sdl_rpc_plugin::commands::CloseApplicationRequest>
+    CloseApplicationCommandPtr;
+
+namespace {
+const uint32_t kConnectionKey = 2u;
+const uint32_t kCorrelationId = 3u;
+}
+
+MessageSharedPtr CreateCloseAppMessage() {
+  using namespace am;
+  auto message = std::make_shared<SmartObject>(smart_objects::SmartType_Null);
+  auto& params = (*message)[strings::params];
+  params[strings::connection_key] = kConnectionKey;
+  params[strings::function_id] = mobile_apis::FunctionID::CloseApplicationID;
+  return message;
+}
+
+class CloseApplicationRequestTest
+    : public CommandRequestTest<CommandsTestMocks::kIsNice> {
+ public:
+  CloseApplicationRequestTest()
+      : mock_app_(CreateMockApp())
+      , message_(CreateCloseAppMessage())
+      , response_message_(CreateMessage()) {}
+
+  void DeffaultRequestRunCheck(const bool success_send_to_hmi) {
+    EXPECT_CALL(app_mngr_, application(kConnectionKey))
+        .WillOnce(Return(mock_app_));
```

Comment:
@IGapchuk Please, double check this **Return**, since in **SetUp** section there is already **ON_CALL** which returns **mock_app_** by default